plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/BlockLocation.java

```diff
@@ -14,22 +14,26 @@
 package io.trino.plugin.hive.fs;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Interner;
+import com.google.common.collect.Interners;
 import io.trino.filesystem.FileEntry.Block;
 
 import javax.annotation.Nullable;
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Arrays.stream;
 import static java.util.Objects.requireNonNull;
 
 public class BlockLocation
 {
+    private static final Interner<String> HOST_INTERNER = Interners.newWeakInterner();
```

Comment:
I think we generally avoid using weak-valued caches because this brings less predictibility for the memory management. So this small single line is a big change to me.

however, what about scoping the interner in the caller, eg in `InternalHiveSplitFactory`?

Reply:
cc @electrum @dain @losipiuk @arhimondr

Reply:
> I think we generally avoid using weak-valued caches because this brings less predictibility for the memory management. So this small single line is a big change to me.

The assumption is there won't be many hosts in the cluster so it should be negligible in practice.

> however, what about scoping the interner in the caller, eg in InternalHiveSplitFactory?

I'm not sure what would that change. For example `CachingDirectoryLister` has global scope and it keeps `BlockLocation` indirectly.
I wanted to avoid extra complexity here since the number of hosts should be small anyway

Reply:
> The assumption is there won't be many hosts in the cluster so it should be negligible in practice.

good point. can you add this as a code comment?

Reply:
not a stupid question, since I was about to ask the same one :P